python/pyspark/ml/connect/__init__.py

```diff
@@ -16,6 +16,9 @@
 #
 
 """Spark Connect Python Client - ML module"""
+from pyspark.sql.connect.utils import check_dependencies
```

Comment:
and we check the required dependency at the top level of module when it's imported.